mlflow/pyfunc/__init__.py

```diff
@@ -1675,6 +1677,38 @@ def _validate_function_python_model(python_model):
                 error_code=INVALID_PARAMETER_VALUE,
             )
 
+    if isinstance(python_model, PythonModel):
+        try:
+            init_has_model_param = "model" in inspect.signature(python_model.__init__).parameters
+            init_has_model_assignment = "self.model =" in inspect.getsource(python_model.__init__)
```

Comment:
This could potentially create a false positive indicator. For instance, what if I initialized the object with:

```python 
class MyModel(mlflow.pyfunc.PythonModel):

    def __init__():
        self.model = None

    def load_context(self, context):
        self.model = <load from src>

    def predict(self, context, model_input, params=None):
        ...
```